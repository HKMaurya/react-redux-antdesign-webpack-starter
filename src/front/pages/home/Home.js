// @flow

// #reigon imports
import React, { PureComponent } from 'react';
import { Button, Icon } from 'antd';
import AnimatedView from '../../components/animatedView/AnimatedView';
import { type RouterProps } from '../../types/react-router';
// #endregion

// #region flow types
export type Props = { ...any } & RouterProps;
export type State = { ...any };
// #endregion

class Home extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    return (
      <AnimatedView>
        <div>
          <h1>ReactJS + Redux + Ant Design</h1>
          <h2>with Hot Reload!!!</h2>
          <h2>and React Router v4</h2>
          <h2>and webpack 4.x</h2>
          <h1>Starter</h1>
          <p>
            <Button type="primary" onClick={this.handlesOnGoAbout}>
              <Icon type="info-circle-o" />
              go to about
            </Button>
          </p>
        </div>
      </AnimatedView>
    );
  }
  // #endregion

  // #region on about button click
  handlesOnGoAbout = (event: SyntheticEvent<>): void => {
    if (event) {
      event.preventDefault();
    }

    const { history } = this.props;
    history.push('/about');
  };
  // #endregion
}

export default Home;
