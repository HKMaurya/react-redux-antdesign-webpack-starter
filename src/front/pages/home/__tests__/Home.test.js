// @flow

// #region imports
import React from 'react';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import Home from '../Home';
// #endregion

// NOTE: react-test-renderer won't work if component uses ref (like antd Button and Icon)
// ONLY solution: mock these components
jest.mock('antd/lib/button', () => () => <button>mocked button</button>);
jest.mock('antd/lib/icon', () => <i>icon</i>);

describe('Home page', () => {
  it('renders as expected', () => {
    const props = {};
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <Home {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
