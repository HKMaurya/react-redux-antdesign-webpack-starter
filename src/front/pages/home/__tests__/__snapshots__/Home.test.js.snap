// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders as expected 1`] = `
<div>
  <section
    className="content view-enter"
  >
    <div>
      <h1>
        ReactJS 16 + Redux + Ant Design
      </h1>
      <h2>
        with Hot Reload!!!
      </h2>
      <h2>
        and React Router v4
      </h2>
      <h2>
        and webpack 4.x
      </h2>
      <h1>
        Starter
      </h1>
      <p>
        <button>
          mocked button
        </button>
      </p>
    </div>
  </section>
</div>
`;
